bin/stackprof

```diff
@@ -18,11 +18,7 @@ parser = OptionParser.new(ARGV) do |o|
   o.on('--graphviz', "Graphviz output (use with dot)"){ options[:format] = :graphviz }
   o.on('--node-fraction [frac]', OptionParser::DecimalNumeric, 'Drop nodes representing less than [frac] fraction of samples'){ |n| options[:node_fraction] = n }
   o.on('--stackcollapse', 'stackcollapse.pl compatible output (use with stackprof-flamegraph.pl)'){ options[:format] = :stackcollapse }
-  o.on('--flamegraph', "timeline-flamegraph output (js)"){ options[:format] = :flamegraph }
-  o.on('--flamegraph-viewer [f.js]', String, "open html viewer for flamegraph output\n\n"){ |file|
-    puts("open file://#{File.expand_path('../../lib/stackprof/flamegraph/viewer.html', __FILE__)}?data=#{File.expand_path(file)}")
-    exit
-  }
+  o.on('--flamegraph', "open a viewer for the flamegraph of the given profile"){ options[:format] = :flamegraph }
```

Comment:
I am personally :-1: on this options consolidation change (not that I have any real pull in the final say...)

Reason being is I have used this to integrate with other repos, and having this in two separate steps is kind of ideal when you want to just save it to a path your your choosing (currently this forces the use of `Tmpdir`), and open at your convenience.  Allowing you to choose a dir also allows you to organize your samples, even when using the private `Stackprof.print_flamegraph` interface, and now that is not available.

Reply:
To maybe meet halfway:

Could you keep the options that exist currently, but change `--flamegraph-viewer` to optionally accept a `file`, and if one isn't provided, it will run a form of the `Tmpdir` code you currently have and open the file.

Reply:
Thanks for the feedback! I don't have strong opinions here, and happy to do whatever would yield the best user experience here.

Open to guidance for what the commandline flags should be here and what the semantics should be.

> and having this in two separate steps is kind of ideal when you want to just save it to a path your your choosing
 
My understand is that if you want to collect multiple profiles for later viewing, you can collect them as the raw stackprof output into a directory of your choosing, then run the `--flamegraph` command to view them. I'm not sure what you would want to do with the intermediate form here other than open them to view as a flamegraph.

The intermediate form is a JavaScript ball that isn't usable (AFAIK) from anything except the flamegraph viewer.

Reply:
> The intermediate form is a JavaScript ball that isn't usable (AFAIK) from anything except the flamegraph viewer.

I always kind of thought of the `--flamegraph` flag command as the "compile" step of the flamegraph, and the `--flamegraph-viewer` flag as the execution.  The "compile" step could be slow, depending on how big your stackprof blob is (you were talking about 100MB blobs, and I know the feeling myself), you might not want to repeat that process if you are looking at data over time, or with a before and after.

Sidenote (and I am _**NOT**_ suggesting doing this in this PR), in regards to the "Javascript ball" being unusable, I have wondered if it made sense to have a option flag to make it a single `.html` file artifact instead of javsacript file but inlining the scripts (so `speedscope` in this case).  Seems like it would avoid the whole `viewer` issue, and it is what the original `perl` version of the `flamegraph` does as well by making it an SVG.

Reply:
I don't have any strong feelings about this change, but I'd prefer not to break people's workflow.